CI/bench/evaluate.py

```diff
@@ -1,142 +1,250 @@
-    for _, row in df.iterrows():
-        if ("master" not in row['framework']) and (master_win_rate >= row['winrate']):
-            pr_comment = ""
-            pr_comment = f"\nBenchmark Test Result - Fail\nEvaluation Results Path: s3://autogluon-ci-benchmark/evaluation/{module_name}/{branch_name}\n"
-
-    with open("final_eval.txt", "w") as file:
-        file.write(pr_comment)
+def main():
+
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument(
+        "--config_path", help="path to generated config path to fetch benchmark name", type=str, required=True
+    )
+    parser.add_argument("--module_name", help="module on which we run benchmark", type=str, required=True)
+    parser.add_argument("--time_limit", help="time limit of the benchmark run", type=str, required=True)
+    parser.add_argument("--branch_name", help="if it happens to be master then just push the cleaned result, do not evaluate", type=str, required=True)
+    parser.add_argument("--benchmark_type", help="type of benchmark to run, tabular, timeseries, automm-text etc.", type=str, required=True)
+
+
+    args = parser.parse_args()
+
+    config_path = args.config_path
+    module_name = args.module_name
+    time_limit = args.time_limit
+    branch_name = args.branch_name
+    benchmark_type = args.benchmark_type
+    df1 = pd.DataFrame()
+    df2 = pd.DataFrame()
+
+    try:
+        for root, dirs, files in os.walk(config_path):
+            for file in files:
+                if file == f"{module_name}_cloud_configs.yaml":
+                    config_file = os.path.join(root, file)
+                    break
+
+        with open(config_file, "r") as f:
+            config = yaml.safe_load(f)
+            benchmark_name = config["benchmark_name"]
+
+        subprocess.run(
+            [
+                "agbench",
+                "aggregate-amlb-results",
+                "autogluon-ci-benchmark",
+                module_name,
+                benchmark_name,
+                "--constraint",
+                time_limit,
+            ],
+            check=True,
+        )
+
+        subprocess.run(
+            [
+                "agbench",
+                "clean-amlb-results",
+                benchmark_name,
+                f"--results-dir-input",
+                f"s3://autogluon-ci-benchmark/aggregated/{module_name}/{benchmark_name}/",
+                "--file-prefix",
+                f"results_automlbenchmark_{time_limit}",
+                "--benchmark-name-in-input-path",
+                "--results-dir-output",
+                "./results",
+            ],
+            check=True,
+        )
+
+        subprocess.run(
+            [
+                "sh",
+                "-c",
+                "rm -f ./results/*_min.csv ./results/*_min.parquet ./results/*.parquet",
```

Comment:
If these are removed, is the logic `if file.endswith(".csv") and not file.endswith("_min.csv") and not file.endswith(".parquet"):` still necessary?

Reply:
Adding this for robustness in case evaluation scripts are changed and more files are added.